islandora.module

```diff
@@ -1685,8 +1690,12 @@ function islandora_object_access($op, $object, $user = NULL) {
  */
 function islandora_islandora_object_access($op, $object, $user) {
   module_load_include('inc', 'islandora', 'includes/utilities');
-
-  return islandora_namespace_accessible($object->id) && user_access($op, $user);
+  if ($object->state != 'A') {
+    return islandora_namespace_accessible($object->id) && user_access($op, $user) && user_access(ISLANDORA_ACCESS_INACTIVE_AND_DELETED_OBJECTS, $user);
```

Comment:
@jordandukart @willtp87 @fuel37 I took out the update hook that adds permissions, as @jonathangreen  suggested. It didn't actually preserve existing functionality - not in the way that you'd been requesting. Which leaves us with a pull request that breaks precedent (apparently) by changing the Islandora configuration footprint. I still think it should pass. It's a feature requested by the community, and addresses a reported security hole in which access management was not meeting user expectations.